lms/djangoapps/course_api/serializers.py

```diff
@@ -32,6 +32,38 @@ def get_uri(self, course_overview):
         return getattr(course_overview, self.uri_attribute)
 
 
+class _AbsolutMediaSerializer(_MediaSerializer):  # pylint: disable=abstract-method
+    """
+    Nested serializer to represent a media object and its absolute path.
+    """
+    requires_context = True
+
+    def __call__(self, serializer_field):
+        self.context = serializer_field.context
+        return super(self).__call__(serializer_field)
+
+    url = serializers.SerializerMethodField(source="*")
+
+    def get_url(self, course_overview):
+        """
+        Convert the media resource's URI to an absolute URL
+        """
+        uri = getattr(course_overview, self.uri_attribute)
+
+        if not uri:
+            return
+
+        url = course_overview.apply_cdn_to_url(uri)
+        field = AbsoluteURLField()
+
+        # In order to use the AbsoluteURLField to have the same
+        # behaviour what ImageSerializer provides, we need to set
+        # the request for the field
+        field._context = {"request": self.context.get("request")}
```

Comment:
I believe this hack is required because you're declaring the field within this method, instead of as a member of this serializer.

I think that if you just declare the field like this:

```python
class _AbsolutMediaSerializer(_MediaSerializer):
    requires_context = True
    url = AbsoluteURLField(source="get_url")

    def get_url(self):
        ...
```
then it should work without `context` hacks.

Reply:
Unfortunately, this cannot be done, because the `BaseSerializer` does not set the `self.initial_data` if data argument was empty. Although serializers has a `get_initial` method, the method will return an empty `OrderedDict` since the serializer did not receive data at all. So we need to pass the context around, but I'm opened to other ideas as well. When I did a research in this topic when I opened the PR, I did not find any other solution which is cleaner.